### PR TITLE
Handle DB errors in whosonline module

### PR DESCRIPTION
**Pull request by alikon**
updated to handle DB errors like #6612 ,#6632  and for #6591

Testing info
simulate a db error 
set the db offline ...

### CODE REVIEW

File: modules/mod_whosonline/helper.php [hunk 1]
```diff
@@ -38,7 +38,14 @@ public static function getOnlineCount()
...
 		$db->setQuery($query);
 
-		$sessions = (array) $db->loadObjectList();
+		try
+		{
+			$sessions = (array) $db->loadObjectList();
+		}
+		catch (RuntimeException $e)
+		{
+			// Don't worry be happy
```
Comment (mbabker) [inline modules/mod_whosonline/helper.php]: Probably want to set `$sessions = 0;` here otherwise the if statement below won't be too happy :-)

Comment (nonumber) [inline modules/mod_whosonline/helper.php]: How so? The database query outputs an array (ObjectList).
Code would be fine dealing with an empty array...

**Comment by nonumber** [comment]
@test success

File: modules/mod_whosonline/helper.php (hunk 1 above)
Comment (mbabker) [inline modules/mod_whosonline/helper.php]: If an exception is thrown, `$sessions` won't be set so the next if statement will throw a notice about an undefined variable.

**Comment by davdebcom** [comment]
@test
Works. <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/6665">issues.joomla.org/joomla-cms/6665</a>.</sub>

**Comment by nonumber** [comment]
Not sure if this should get merged with regards to discussion on https://github.com/joomla/joomla-cms/pull/6710

**Comment by dgt41** [comment]
@test ok

**Comment by brianteeman** [comment]
Setting RTC - thanks<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/6665">issues.joomla.org/joomla-cms/6665</a>.</sub>

**Pull request closed without merging**